core/src/avm1/activation.rs

```diff
@@ -806,6 +806,7 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         let fn_name_value = self.context.avm1.pop();
         let fn_name = fn_name_value.coerce_to_string(self)?;
         let num_args = self.context.avm1.pop().coerce_to_f64(self)? as usize; // TODO(Herschel): max arg count?
+        let num_args = num_args.min(self.context.avm1.stack.len());
```

Comment:
From testing, `num_args` seems like it should use `coerce_to_u32` (wraps at `4294967297`). This also means that `-1` ends up acting like `u32::MAX` and would pop all values off of the stack.